django_app/redbox_app/redbox_core/models.py

```diff
@@ -42,16 +43,63 @@ class ProcessingStatusEnum(models.TextChoices):
     complete = "complete"
 
 
-# TO DO: Based on /redbox/models/file.py, but not complete
-class File(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    path = models.TextField(help_text="location of file")
-    name = models.TextField()
-    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, default=ProcessingStatusEnum.uploaded)
+class FileTypeEnum(models.TextChoices):
+    eml = ".eml"
+    html = ".html"
+    json = ".json"
+    md = ".md"
+    msg = ".msg"
+    rst = ".rst"
+    rtf = ".rtf"
+    txt = ".txt"
+    xml = ".xml"
+    csv = ".csv"
+    doc = ".doc"
+    docx = ".docx"
+    epub = ".epub"
+    odt = ".odt"
+    pdf = ".pdf"
+    ppt = ".ppt"
+    pptx = ".pptx"
+    tsv = ".tsv"
+    xlsx = ".xlsx"
+    htm = ".htm"
+
+
+class File(UUIDPrimaryKeyBase, TimeStampedModel):
+    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, null=False, blank=False)
+    original_file = models.FileField(storage=settings.BUCKET_NAME)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    @property
+    def file_type(self):
+        name = self.original_file.name
+        return name.split(".")[-1]
+
+    @property
+    def file_url(self):
+        return self.original_file.url
+
+    @property
+    def name(self):
+        return self.original_file.name
 
     def get_processing_text(self) -> str:
         processing_status_list = list(ProcessingStatusEnum)
         stage = processing_status_list.index(self.processing_status)
         if stage == len(processing_status_list) - 1:
             return self.processing_status
         return f"{stage + 1}/{len(processing_status_list) - 1} {self.processing_status}"
+
+
+class ChatHistory(UUIDPrimaryKeyBase, TimeStampedModel):
+    name = models.TextField(max_length=1024, null=False, blank=False)
+    users = models.ManyToManyField(User)
+    files_received = models.ForeignKey(File, on_delete=models.CASCADE, related_name="files_received")
```

Comment:
this is a tough one.... i feel that the relationship between a chat and a file is many-to-many... and perhaps more complex than that... maybe need a through model to label the collection of files, e.g. 'environment" may group files 1, 2, & 3 and "industry" might group file 3,4,5... just a thought, open to discussion

Reply:
What does files_received in a chat even mean in the first place? Do users add files to certain chats? I thought it would be more of a case of files_selected for chats, and files_referenced for a message?

Reply:
i dont know tbh. @lmwilkigov do you know?

Reply:
> I thought it would be more of a case of files_selected for chats, and files_referenced for a message?
This sounds broadly right. If this is the case, I imagine we don't need a list of files_referenced for a whole history that combines the message files_referenced.

* @lmwilkigov I think this also leads to a question about whether we have a blanket search on all files a user has uploaded, or they have the option to choose specific ones?

Reply:
Are we happy with the changes I've made regarding this?

Reply:
> It looks like all files from recent meetings, but then you'd still need to keep track of the files for a specific chat in case the user uploads more files after-the-fact?

I'm not certain we would? files_selected would future proof us though.

Reply:
I think it depends on if we're going to limit the RAG search to the files present in the chat.

I have a chat that has files a, b, c.

I chat with that for a while.

I then go upload files d and e.

I make a new chat and chat there for a bit.

If we have chat history, and I can go back and view the first chat, without saving the files used for that chat there's no way to know what files that chat has access to.

But if we don't limit the RAG to those files anyway, then it doesn't really make a difference because as soon as I enter a new message into chat 1 it now also includes d and e in all new retrievals

Reply:
I don't think we limit to those files for MVP, so it will include d and e. But it sounds like we will want to in the future. So it depends whether it's easiest to add it in now or wait until later.